### PR TITLE
fixed #125: unfinished downloads prevent new download on next try

### LINKED ISSUE #125: If a download is interrupted, unfinished files will prevent a new download
Problem:
---

If you interrupt the application while downloading files (for example with CTRL-C), the unfinished downloads will stay in the destination directory. If you start the download again, the script will find a file with the expected name, but can not recognize, that this is not a fully downloaded ebook or code file.

Solution:
---

Start the download with a temporary file and after the file is fully downloaded, rename the file to the destination file name. If an exception occurs, clean up the temporary file.

### PR DESCRIPTION
**Pull request by awegmann**
I added the code to create downloads with temp files. The block `with open....` is surrounded with a `try-finally` to make sure, that the file is correctly removed in case of an exception or interrupt. 

BTW: 
1. I added `.idea` to `.gitignore` because I'm using PyCharm
~~2. I added `'api'` in `setup.py` without deeper knowledge of this mechanism, but after cloning and installing in a virtual environment, I got errors, because `api` module was missing.~~

### CODE REVIEW

**Comment by mjenczmyk** [comment]
Could you rebase with this repository `master` and squash your commits into single one?

File: src/packtPublishingFreeEbook.py [hunk 1]
```diff
@@ -197,6 +197,7 @@ def get_product_download_urls(product_id):
                     else:
                         target_download_path = os.path.join(self.cfg.download_folder_path)
                     full_file_path = os.path.join(target_download_path, '{}.{}'.format(file_name, file_extention))
+                    temp_file_path = os.path.join(target_download_path, 'download.tmp')
```
Comment (mjenczmyk) [inline src/packtPublishingFreeEbook.py]: There is a `tempfile` package in standard library - maybe we could download to temporary directory first, and than move to expected download path? It's just an idea.

**Comment by awegmann** [comment]
fixed conflicts, offer you to merge...

**Comment by mjenczmyk** [comment]
All right, I'll test it and then squash and rebase.

**Comment by mjenczmyk** [comment]
Thanks, I've checked that it works and merged it.

**Pull request merged**